### PR TITLE
chore: remove checkstyle rule

### PR DESCRIPTION
**Pull request by monperrus**
Too much convention is useless and no fun.
Julien Duribreux (@?) was right, we have to do something.

### CODE REVIEW

**Comment by msteinbeck** [comment]
I prefer having strict rules :).

**Comment by monperrus** [comment]
I also prefer strict rules.

The question is whether at some point too strict becomes counter-productive.

File: checkstyle.xml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.2//EN" "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
 <module name="Checker">
-  <module name="NewlineAtEndOfFile">
-    <property name="lineSeparator" value="lf"/>
-  </module>
```
Comment (surli) [inline checkstyle.xml]: I'll keep this one: keeping a line at the end of a file is better for the diff, and at least we ensure that we won't have windows-style (or mac style ;) ) end of lines.

See: http://checkstyle.sourceforge.net/config_misc.html#NewlineAtEndOfFile

**Comment by monperrus** [comment]
Interesting. `NewlineAtEndOfFile` only checks the last line.

I discover that we have a mix of \\r\\n and \n in Spoon :-(  That's bad.

What's the checkstyle rule to check this and enforce we only have \\n?

**Comment by surli** [comment]
> What's the checkstyle rule to check this and enforce we only have \n?

I found that:
```
<module name="RegexpMultiline">
    <property name="format" value="\r\n"/>
    <property name="message" value="Do not use Windows line endings"/>
</module>
```

Source: https://github.com/checkstyle/checkstyle/issues/893

**Comment by monperrus** [comment]
awesome. would you make a PR to add it and modify the violations accordingly?

**Comment by surli** [comment]
> would you make a PR to add it and modify the violations accordingly?

OK

**Comment by zielint0** [comment]
I found that for Git developers git configuration during setup is important: fetch unix, commit unix.

**Pull request merged**